odxtools/parameters/tablestructparameter.py

```diff
@@ -61,15 +60,31 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
             self._table_key = odxlinks.resolve(self.table_key_ref, TableKeyParameter)
 
     @override
+    @final
     def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
```

Comment:
We should consider in the future making `_resolve_snrefs` take a "context" or "lookup" object, where diag_layer is a property.

We are getting too many "_resolve_snrefs" special cases

Reply:
agreed. this is a major refactoring, though. I'll open an issue...

Reply:
ok, done: https://github.com/mercedes-benz/odxtools/issues/290